java/src/main/java/com/google/cloud/dataproc/templates/databases/MongoToBQ.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.dataproc.templates.databases;
+
+import static com.google.cloud.dataproc.templates.util.TemplateConstants.*;
+
+import com.google.cloud.dataproc.templates.BaseTemplate;
+import java.sql.SQLException;
+import java.util.concurrent.TimeoutException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.streaming.StreamingQueryException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MongoToBQ implements BaseTemplate {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(MongoToBQ.class);
+
+  private final String projectId;
+  private final String inputURI;
+  private final String inputDatabase;
+  private final String inputCollection;
+  private final String outputMode;
+  private final String bqOutputDataset;
+  private final String bqOutputTable;
+  private final String bqTempBucket;
+
+  public MongoToBQ() {
+
+    projectId = getProperties().getProperty(PROJECT_ID_PROP);
+    inputURI = getProperties().getProperty(MONGO_BQ_INPUT_URI);
```

Comment:
Validation missing for all the input values (refer SpannerToGCSConfig.java or CassandraToBqConfig.java)